catalogue/views.py

```diff
@@ -121,3 +121,46 @@ class ComparisonView(FilterView):
     template_name = "catalogue/compare.html"
     context_object_name = "components"
     filterset_class = ComponentComparisonFilter
+
+
+class ReportView(generic.detail.SingleObjectMixin, generic.edit.FormView):
+    template_name = "catalogue/modals/report/form.html"
+    success_template = "catalogue/modals/report/success.html"
+    form_class = ReportForm
+    model = Component
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
```

Comment:
is this necessary?

Reply:
Seems to be necessary. When I remove it, it doesn't work anymore. Holds for post and get. The super class seems to need this attribute. Also tried to remove the post and get methods completely.
Message: 
  File "/Users/ironmonkeyapps/Desktop/Studium/Master/Semester2/solutionCatalogue/catalogue/views.py", line 156, in get_context_data
    context = super().get_context_data(**kwargs)
  File "/Users/ironmonkeyapps/.local/share/virtualenvs/solutionCatalogue--U9CtuRt/lib/python3.9/site-packages/django/views/generic/detail.py", line 94, in get_context_data
    if self.object:
AttributeError: 'ReportView' object has no attribute 'object'

Reply:
alright. 

> The base implementation of this method requires that the self.object attribute be set by the view (even if None). Be sure to do this if you are using this mixin without one of the built-in views that does so.

https://docs.djangoproject.com/en/3.2/ref/class-based-views/mixins-single-object/#django.views.generic.detail.SingleObjectMixin.get_context_data